nemo/collections/nlp/data/machine_translation/preproc_mt_data.py

```diff
@@ -756,63 +765,51 @@ def train_tokenizers(
-                        if encoder_tokenizer_name == "yttm":
-                            yttm.BPE.train(
-                                data=concat_data_path,
-                                vocab_size=encoder_tokenizer_vocab_size,
-                                model=os.path.join(out_dir, encoder_tokenizer_model),
-                                coverage=encoder_tokenizer_coverage,
-                                n_threads=-1,
-                            )
-                        else:
-                            create_spt_model(
-                                data_file=concat_data_path,
-                                vocab_size=encoder_tokenizer_vocab_size,
-                                sample_size=encoder_training_sample_size,
-                                do_lower_case=False,
-                                tokenizer_type='bpe',
-                                character_coverage=encoder_tokenizer_coverage,
-                                output_dir=out_dir,
-                                bos=True,
-                                eos=True,
-                                pad=True,
-                                control_symbols=spt_symbols,
-                                user_defined_symbols=encoder_special_tokens,
-                            )
-                            os.rename(
-                                os.path.join(out_dir, 'tokenizer.model'),
-                                os.path.join(out_dir, encoder_tokenizer_model),
-                            )
+                decoder_tokenizer_model = encoder_tokenizer_model
+                if global_rank == 0:
+                    if os.path.isfile(encoder_tokenizer_model):
+                        logging.info(
+                            f'Shared tokenizer model {encoder_tokenizer_model} already exists. Remove file if training a new tokenizer model.'
+                        )
+                    else:
+                        logging.info(
+                            f'Shared tokenizer model {encoder_tokenizer_model} not found. Training tokenizer model.'
+                        )
+                        with tempfile.TemporaryDirectory() as tmp:
+                            concat_data_path = os.path.join(tmp, 'concat_dataset.txt')
+                            os.system('cat %s %s > %s' % (src_fname, tgt_fname, concat_data_path))
+                            if encoder_tokenizer_name == "yttm":
+                                yttm.BPE.train(
+                                    data=concat_data_path,
+                                    vocab_size=encoder_tokenizer_vocab_size,
+                                    model=os.path.join(out_dir, encoder_tokenizer_model),
+                                    coverage=encoder_tokenizer_coverage,
+                                    n_threads=-1,
+                                )
+                            else:
```

Comment:
Do we have only yttm and sentencepiece supported here?
(why not if yttp: elif spt: else: raise Error ?

Reply:
This is required to support some of our other scripts like the one to create tarred dataset. Changing other scripts is a larger workload, changing this function's behavior to return None when things can't be trained handles everything without changing other scripts

Reply:
Ok let's keep it as it was before.